src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,101 @@
+package duke.storage;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Scanner;
+
+import duke.exceptions.LoadException;
+import duke.tasks.Deadlines;
+import duke.tasks.Events;
+import duke.tasks.Task;
+import duke.tasks.Todos;
+
+
+
+
+/**
+ * A class which handles reading and writing data onto a file.
+ * This is needed for saving the task list.
+ */
+public class Storage {
+    private final String filePath;
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Reads and loads the data from data.txt.
+     * Each line is read and a corresponding task is created.
+     * From there, a new task list is constructed.
+     * @return A list of tasks.
+     * @throws FileNotFoundException If file is not found.
+     */
+    public TaskList loadFile() throws LoadException {
+
+        ArrayList<Task> loadData;
```

Comment:
Perhaps a more intuitive variable name can be used for `loadData`
```suggestion
        ArrayList<Task> loadedDataList;
```